.github/workflows/main.yml

```diff
@@ -13,27 +13,130 @@ on:
 
 jobs:
   build:
-    runs-on: ${{ matrix.os }}
-
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+    defaults:
+      run:
+        shell: ${{ matrix.config.shell }}
     strategy:
+      fail-fast: false
       matrix:
-        compiler: [clang, gcc]
-        os: [ubuntu-latest]
+        config:
+          - name: Linux GCC
+            os: ubuntu-latest
+            compiler: gcc
+            shell: bash
+
+          - name: macOS Clang
+            os: macos-latest
+            compiler: clang
+            shell: bash
+
+          - name: MSYS2 MINGW64
+            os: windows-latest
+            compiler: gcc
+            shell: 'msys2 {0}'
+            msystem: mingw64
+            msys-env: mingw-w64-x86_64
+
+          - name: MSYS2 MINGW32
+            os: windows-latest
+            compiler: gcc
+            shell: 'msys2 {0}'
+            msystem: mingw32
+            msys-env: mingw-w64-i686
 
     steps:
-      - name: Install dependencies
-        run: sudo apt-get update && sudo apt-get install libfluidsynth-dev libpng-dev libsamplerate0-dev libsdl2-dev libsdl2-image-dev libsdl2-mixer-dev libsdl2-net-dev ninja-build
+      - name: Install pandoc
+        if: runner.os == 'Windows'
+        shell: cmd
+        run: choco install pandoc
+
+      - name: Install dependencies (Linux)
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt-get update
+          sudo apt-get install \
+            libsdl2-dev \
+            libsdl2-mixer-dev \
+            libsdl2-net-dev \
+            libpng-dev \
+            libsamplerate0-dev \
+            libfluidsynth-dev
+
+      - name: Install dependencies (macOS)
+        if: runner.os == 'macOS'
+        env:
+          HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK: 1
+        run: |
+          brew install \
+            automake \
+            pandoc \
```

Comment:
Is this actually being used anywhere?

Reply:
All documentation is converted to Markdown. I don't know how this works on macOS, on Windows you have to install some program or browser extension to view Markdown documentation. We can update the macOS packaging to use `pandoc` to convert the documentation to `.html`.

Reply:
I think pandoc only used on Windows? 

https://github.com/chocolate-doom/chocolate-doom/blob/7f25bf6fbd0b6917912db86d5e321886375247aa/pkg/win32/GNUmakefile#L28-L53

Reply:
> I think pandoc only used on Windows?

True, but we can add this code to macOS packaging as well. Only I'm not sure if we should do that, how does a macOS launcher user access the documentation?